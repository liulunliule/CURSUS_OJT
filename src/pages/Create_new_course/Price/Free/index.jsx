import React from "react";
import "./index.scss";
import { Switch } from "antd";
function Free_page() {
    return (
        <div className="free_page">
            <div className="free_page_switch-btn">
                <Switch style={{ marginRight: "10px" }} />
                Require Log in
            </div>

            <div className="free_page_switch-btn">
                <Switch style={{ marginRight: "10px" }} />
                Require Enroll
            </div>

            <p className="free_page-content">
                If the course is free, if student require to enroll your course,
                if not required enroll, if students required sign in to your
                website to take this course.
            </p>
        </div>
    );
}

export default Free_page;
