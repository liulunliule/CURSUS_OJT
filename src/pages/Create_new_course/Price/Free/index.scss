.free_page {
    margin-top: 34px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    .free_page_switch-btn {
        margin-bottom: 17px;
        color: #231f20;
        font-size: 14px;
        font-weight: 500;
    }

    .free_page-content {
        margin-top: 20px;
        text-align: center;
        padding: 0 150px;
    }
}

.ant-switch.ant-switch-checked {
    background-color: #ed2a26;
}
