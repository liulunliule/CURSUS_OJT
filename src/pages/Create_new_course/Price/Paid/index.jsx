import React from "react";
import "./index.scss";
function Paid_page() {
    return (
        <div className="paid_page">
            <div className="paid_page-input">
                <label htmlFor="">Regular Price*</label>
                <form action="" className="paid_page-form">
                    <input type="text" name="" id="" placeholder="$0" />
                    <span>USE</span>
                </form>
            </div>
            <div className="paid_page-input">
                <label htmlFor="">Regular Price*</label>
                <form action="" className="paid_page-form">
                    <input type="text" name="" id="" placeholder="$0" />
                    <span>USE</span>
                </form>
            </div>
        </div>
    );
}

export default Paid_page;
