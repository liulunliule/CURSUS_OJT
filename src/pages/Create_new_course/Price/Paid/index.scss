.paid_page {
    margin-top: 30px;

    &-input {
        margin-bottom: 30px;
        label {
            font-size: 14px;
            font-weight: 500;
            margin-bottom: 10px;
            color: #333;
        }
    }

    &-form {
        position: relative;
        display: flex;
        width: 33.33333333%;
        background-color: #fff;
        border: 0.8px solid #e5e5e5;
        border-radius: 4px;
        color: #333;
        font-size: 14px;
        padding: 9.5px 28px 9.5px 14px;
        input {
            width: 100%;
            border: none;
            outline: none;

            &::placeholder {
                color: #c4c1c1;
            }
        }

        span {
            color: #333;
            font-weight: 600;
            position: absolute;
            background: #f7f7f7;
            right: 5px;
            top: 5px;
            min-height: 30px;
            padding: 0 12px;
            line-height: 30px;
            border-radius: 3px;
        }
    }
}
