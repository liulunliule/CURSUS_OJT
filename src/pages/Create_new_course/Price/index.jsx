import React, { useEffect, useState } from "react";
import "./index.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCartShopping,
  faMoneyBill,
  faTag,
} from "@fortawesome/free-solid-svg-icons";
import Free_page from "./Free";
import Paid_page from "./Paid";

function Price({ price, setPrice }) {
  const [selectedOption, setSelectOption] = useState(
    price > 0 ? "paid" : "free"
  );

  const handlePriceChange = (newPrice) => {
    setPrice(newPrice);
    setSelectOption("paid");
  };
  useEffect(() => {
    if (selectedOption === "free") {
      setPrice(0);
    }
  }, [selectedOption, setPrice]);
  return (
    <div className="create_course_price">
      <div className="price_title">
        <FontAwesomeIcon icon={faMoneyBill} className="price_icon" />
        Price
      </div>
      <div className="price_form">
        <div className="price_link-list">
          <button
            className={`price_link ${
              selectedOption === "free" ? "active" : ""
            }`}
            onClick={() => setSelectOption("free")}
          >
            <FontAwesomeIcon
              icon={faTag}
              style={{ marginRight: "5px", fontSize: "18px" }}
            />
            Free
          </button>
          <button
            className={`price_link ${
              selectedOption === "paid" ? "active" : ""
            }`}
            onClick={() => setSelectOption("paid")}
          >
            <FontAwesomeIcon
              icon={faCartShopping}
              style={{ marginRight: "5px" }}
            />
            Paid
          </button>
        </div>
        {selectedOption === "free" && <Free_page />}
        {selectedOption === "paid" && (
          <Paid_page price={price} onPriceChange={handlePriceChange} />
        )}
      </div>
    </div>
  );
}

export default Price;
