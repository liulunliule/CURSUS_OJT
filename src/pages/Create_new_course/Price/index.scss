.create_course_price {
    .price_title {
        padding: 19px 0;
        border-top: 1px solid #efefef;
        border-bottom: 1px solid #efefef;
        display: flex;
        align-items: center;
        font-size: 18px;
        font-family: "Roboto", sans-serif;
        font-weight: 500;
        gap: 15px;

        .price_icon {
            font-size: 20px;
        }
    }

    .price_form {
        background: #fff;
        margin-top: 30px;
        padding: 30px;
        border-radius: 3px;
        border: 1px solid #efefef;

        .price_link-list {
            display: flex;
            justify-content: space-between;
        }
        .price_link {
            flex: 1;
            border: 1px solid #efefef;
            font-size: 14px;
            text-align: center;
            font-weight: 500;
            color: #333;
            background: #f7f7f7;
            padding: 12px;

            &.active {
                color: #fff;
                background-color: #ed2a26;
            }
        }
    }
}
