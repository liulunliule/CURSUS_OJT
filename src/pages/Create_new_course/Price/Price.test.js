import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'; // Import để sử dụng snapshot
import Price from '.';
import Free_page from './Free/index';
import Paid_page from './Paid/index';

// Mock các component Free_page và Paid_page
jest.mock('./Free/index', () => () => <div>Free Page Content</div>);
jest.mock('./Paid/index', () => ({ price, onPriceChange }) => (
  <div>
    <div>Paid Page Content</div>
    <button onClick={() => onPriceChange(200)}>Change Price</button>
  </div>
));

// Test snapshot
describe('Price Component Snapshot', () => {
  test('matches snapshot when price is 0', () => {
    const tree = renderer
      .create(<Price price={0} setPrice={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('matches snapshot when price is greater than 0', () => {
    const tree = renderer
      .create(<Price price={100} setPrice={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Test behavior
describe('Price Component', () => {
  test('renders correctly and shows Free Page content', () => {
    render(<Price price={0} setPrice={() => {}} />);
    expect(screen.getByText('Free Page Content')).toBeInTheDocument();
    expect(screen.queryByText('Paid Page Content')).not.toBeInTheDocument();
    expect(screen.getByText('Free')).toHaveClass('active');
    expect(screen.getByText('Paid')).not.toHaveClass('active');
  });

  test('switches to Paid Page and handles price change', () => {
    const mockSetPrice = jest.fn();
    render(<Price price={100} setPrice={mockSetPrice} />);
    expect(screen.getByText('Paid Page Content')).toBeInTheDocument();
    expect(screen.queryByText('Free Page Content')).not.toBeInTheDocument();
    fireEvent.click(screen.getByText('Free'));
    expect(screen.getByText('Free Page Content')).toBeInTheDocument();
    expect(screen.queryByText('Paid Page Content')).not.toBeInTheDocument();
    fireEvent.click(screen.getByText('Paid'));
    fireEvent.click(screen.getByText('Change Price'));
    expect(mockSetPrice).toHaveBeenCalledWith(200);
  });
});
