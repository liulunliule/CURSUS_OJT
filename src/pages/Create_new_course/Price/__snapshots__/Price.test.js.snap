// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Component Snapshot matches snapshot when price is 0 1`] = `
<div
  className="create_course_price"
>
  <div
    className="price_title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-money-bill price_icon"
      data-icon="money-bill"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 64C28.7 64 0 92.7 0 128L0 384c0 35.3 28.7 64 64 64l448 0c35.3 0 64-28.7 64-64l0-256c0-35.3-28.7-64-64-64L64 64zm64 320l-64 0 0-64c35.3 0 64 28.7 64 64zM64 192l0-64 64 0c0 35.3-28.7 64-64 64zM448 384c0-35.3 28.7-64 64-64l0 64-64 0zm64-192c-35.3 0-64-28.7-64-64l64 0 0 64zM288 160a96 96 0 1 1 0 192 96 96 0 1 1 0-192z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Price
  </div>
  <div
    className="price_form"
  >
    <div
      className="price_link-list"
    >
      <button
        className="price_link active"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-tag "
          data-icon="tag"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "fontSize": "18px",
              "marginRight": "5px",
            }
          }
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 80L0 229.5c0 17 6.7 33.3 18.7 45.3l176 176c25 25 65.5 25 90.5 0L418.7 317.3c25-25 25-65.5 0-90.5l-176-176c-12-12-28.3-18.7-45.3-18.7L48 32C21.5 32 0 53.5 0 80zm112 32a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Free
      </button>
      <button
        className="price_link "
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-cart-shopping "
          data-icon="cart-shopping"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "marginRight": "5px",
            }
          }
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 24C0 10.7 10.7 0 24 0L69.5 0c22 0 41.5 12.8 50.6 32l411 0c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3l-288.5 0 5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5L488 336c13.3 0 24 10.7 24 24s-10.7 24-24 24l-288.3 0c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5L24 48C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Paid
      </button>
    </div>
    <div>
      Free Page Content
    </div>
  </div>
</div>
`;

exports[`Price Component Snapshot matches snapshot when price is greater than 0 1`] = `
<div
  className="create_course_price"
>
  <div
    className="price_title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-money-bill price_icon"
      data-icon="money-bill"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M64 64C28.7 64 0 92.7 0 128L0 384c0 35.3 28.7 64 64 64l448 0c35.3 0 64-28.7 64-64l0-256c0-35.3-28.7-64-64-64L64 64zm64 320l-64 0 0-64c35.3 0 64 28.7 64 64zM64 192l0-64 64 0c0 35.3-28.7 64-64 64zM448 384c0-35.3 28.7-64 64-64l0 64-64 0zm64-192c-35.3 0-64-28.7-64-64l64 0 0 64zM288 160a96 96 0 1 1 0 192 96 96 0 1 1 0-192z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Price
  </div>
  <div
    className="price_form"
  >
    <div
      className="price_link-list"
    >
      <button
        className="price_link "
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-tag "
          data-icon="tag"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "fontSize": "18px",
              "marginRight": "5px",
            }
          }
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 80L0 229.5c0 17 6.7 33.3 18.7 45.3l176 176c25 25 65.5 25 90.5 0L418.7 317.3c25-25 25-65.5 0-90.5l-176-176c-12-12-28.3-18.7-45.3-18.7L48 32C21.5 32 0 53.5 0 80zm112 32a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Free
      </button>
      <button
        className="price_link active"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-cart-shopping "
          data-icon="cart-shopping"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "marginRight": "5px",
            }
          }
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 24C0 10.7 10.7 0 24 0L69.5 0c22 0 41.5 12.8 50.6 32l411 0c26.3 0 45.5 25 38.6 50.4l-41 152.3c-8.5 31.4-37 53.3-69.5 53.3l-288.5 0 5.4 28.5c2.2 11.3 12.1 19.5 23.6 19.5L488 336c13.3 0 24 10.7 24 24s-10.7 24-24 24l-288.3 0c-34.6 0-64.3-24.6-70.7-58.5L77.4 54.5c-.7-3.8-4-6.5-7.9-6.5L24 48C10.7 48 0 37.3 0 24zM128 464a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zm336-48a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Paid
      </button>
    </div>
    <div>
      <div>
        Paid Page Content
      </div>
      <button
        onClick={[Function]}
      >
        Change Price
      </button>
    </div>
  </div>
</div>
`;
